tests/components/sharkiq/test_shark_iq.py

```diff
@@ -0,0 +1,283 @@
+"""Test the Shark IQ vacuum entity."""
+from copy import deepcopy
+import enum
+from typing import Dict, List
+
+from sharkiqpy import AylaApi, Properties, SharkIqAuthError, SharkIqVacuum, get_ayla_api
+
+from homeassistant.components.sharkiq import SharkIqUpdateCoordinator
+from homeassistant.components.sharkiq.vacuum import (
+    ATTR_ERROR_CODE,
+    ATTR_ERROR_MSG,
+    ATTR_LOW_LIGHT,
+    ATTR_RECHARGE_RESUME,
+    STATE_RECHARGING_TO_RESUME,
+    SharkVacuumEntity,
+)
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    SUPPORT_BATTERY,
+    SUPPORT_FAN_SPEED,
+    SUPPORT_LOCATE,
+    SUPPORT_PAUSE,
+    SUPPORT_RETURN_HOME,
+    SUPPORT_START,
+    SUPPORT_STATE,
+    SUPPORT_STATUS,
+    SUPPORT_STOP,
+)
+from homeassistant.config_entries import ConfigEntriesFlowManager, ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from .const import (
+    SHARK_DEVICE_DICT,
+    SHARK_METADATA_DICT,
+    SHARK_PROPERTIES_DICT,
+    TEST_PASSWORD,
+    TEST_USERNAME,
+)
+
+from tests.async_mock import MagicMock, patch
+
+try:
+    import ujson as json
```

Comment:
Why do we try to import this?

Reply:
Unless something has changed recently that I've missed (very possible, lol), `ujson` is significantly faster than the stock `json`, so if it's present, might as well use it.

Reply:
We don't use `ujson` in the project so it won't be present unless you install it manually. We don't want this check in our tests.